cmake/HPX_AddComponent.cmake

```diff
@@ -250,18 +250,11 @@ function(add_hpx_component name)
     endforeach()
   endif()
```

Comment:
Do we still need this part?

Reply:
Removing this works when we add the boost workaround to the `CMAKE_CUDA_FLAGS` :ok_hand: